ddl/ddl_api.go

```diff
@@ -639,10 +639,23 @@ func checkDuplicateColumn(cols []interface{}) error {
 	return nil
 }
 
+func checkIsAutoIncrementColumn(colDefs *ast.ColumnDef) bool {
+	for _, option := range colDefs.Options {
+		if option.Tp == ast.ColumnOptionAutoIncrement {
+			return true
+		}
+	}
+	return false
+}
+
 func checkGeneratedColumn(colDefs []*ast.ColumnDef) error {
 	var colName2Generation = make(map[string]columnGenerationInDDL, len(colDefs))
+	autoIncrementColumns := make(map[string]struct{})
```

Comment:
It seems that only one AUTO_INCREMENT column per table, could we use `autoIncrementColumns` directly?

Reply:
yep, seems so~